addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQCloudHandler.java

```diff
@@ -0,0 +1,322 @@
+     *
+     * @param listener the listener to unregister
+     * @return <code>true</code> if listener was previously registered and is now unregistered; <code>false</code>
+     *         otherwise
+     */
+    public boolean unregisterBedStatusListener(final BedStatusListener listener) {
+
+        boolean result = bedStatusListeners.remove(listener);
+        if (result) {
+            updateListenerManagement();
+        }
+
+        return result;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        // all channels are read-only
+    }
+
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+
+        Collection<ConfigStatusMessage> configStatusMessages = new ArrayList<>();
+
+        SleepIQCloudConfiguration config = getConfigAs(SleepIQCloudConfiguration.class);
+        String username = config.username;
+        String password = config.password;
+
+        if (username == null || username.isEmpty()) {
+            configStatusMessages.add(ConfigStatusMessage.Builder.error(USERNAME)
+                    .withMessageKeySuffix(SleepIQConfigStatusMessage.USERNAME_MISSING.getMessageKey())
+                    .withArguments(USERNAME).build());
+        }
+
+        if (password == null || password.isEmpty()) {
+            configStatusMessages.add(ConfigStatusMessage.Builder.error(PASSWORD)
+                    .withMessageKeySuffix(SleepIQConfigStatusMessage.PASSWORD_MISSING.getMessageKey())
+                    .withArguments(PASSWORD).build());
+        }
+
+        return configStatusMessages;
+    }
+
+    /**
+     * Get a list of all beds registered to the cloud service account.
+     *
+     * @return the list of beds (never <code>null</code>)
+     */
+    public List<Bed> getBeds() {
+
+        SleepIQ cloud = getCloud();
+        if (cloud == null) {
+            return Collections.emptyList();
+        }
+
+        return cloud.getBeds();
+    }
+
+    /**
+     * Get the {@link Bed} corresponding to the given identifier.
+     *
+     * @param bedId the bed identifier
+     * @return the identified {@link Bed} or <code>null</code> if no such bed exists
+     */
+    public Bed getBed(final String bedId) {
+
+        for (Bed bed : getBeds()) {
+
+            if (bedId.equals(bed.getBedId())) {
+                return bed;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Update the given properties with attributes of the given bed. If no properties are given, a new map will be
+     * created.
+     *
+     * @param bed the source of data
+     * @param properties the properties to update (this may be <code>null</code>)
+     * @return the given map (or a new map if no map was given) with updated/set properties from the supplied bed
+     */
+    public Map<String, String> updateProperties(final Bed bed, Map<String, String> properties) {
+
+        if (properties == null) {
```

Comment:
Based on the comment above, I have moved this map creation to the caller. This will be included in the fixes I push shortly.